packages/@aws-cdk/aws-codedeploy/lib/traffic-routing-config.ts

```diff
@@ -0,0 +1,159 @@
+import { Duration } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnDeploymentConfig } from './codedeploy.generated';
+
+/**
+ * Represents the structure to pass into the underlying CfnDeploymentConfig class.
+ */
+export interface TrafficRoutingConfig {
+  /**
+   * represents the underlying traffic routing config structure
+   */
+  readonly config: CfnDeploymentConfig.TrafficRoutingConfigProperty;
```

Comment:
Can you either flatten this out into `TrafficRoutingConfig` or add a new interface that represents
the `TrafficRoutingConfigProperty`? We don't like to leak anything from the L1 layer, even if that
means duplicating some stuff.